src/app/malaria-101/activity-1/activity-1-2.component.ts

```diff
@@ -0,0 +1,104 @@
+    public solnArr = ['red-blood-cells.png',
+                       'character-1.png',
+                       'mosquito.png',
+                       'mosquito.png',
+                       'character-2.png',
+                       'plasmodium.png']
+    public labelsArr = ['Plasmodium / Virus',
+                       'First infected mosquito',
+                       'First infected person',
+                       'Infected red blood cells',
+                       'Second infected mosquito',
+                       'Second infected person']
+
+    constructor(private _sharedData: SharedDataService) {
+    }
+
+    ngOnInit() {
+        this._sharedData.customAlert('Drag and drop the images in the container to complete the life cycle', '', 'warning');
+    }
+
+    /**
+     * Utility function used in activity indicator
+     * @param {Number} number Number of activities in a stage - 3 (default)
+     */
+    createRange(number) {
+        const items: number[] = [];
+        for (let i = 1; i <= number; i++) {
+            items.push(i);
+        }
+        return items;
+    }
+
+    allowDrop(ev) {
+        ev.preventDefault();
+    }
+
+    drag(ev) {
+        ev.dataTransfer.setData('text', ev.target.id);
+        if (ev.target.parentNode.id > 0 && ev.target.parentNode.id <= 6) {
+            delete this.currArrState[Number(ev.target.parentNode.id)];
+        }
+    }
+
+    drop(ev) {
+        ev.preventDefault();
+        const data = document.getElementById(ev.dataTransfer.getData('text'));
+        const srcParent = data.parentNode;
+        const tgt = ev.currentTarget.firstElementChild;
+
+        if (tgt) {
+            ev.currentTarget.replaceChild (data, tgt);
+            srcParent.appendChild (tgt);
+            let firstSrc = ev.currentTarget.firstElementChild.src;
+            let secondSrc = tgt.src;
+            firstSrc = firstSrc.substr(firstSrc.lastIndexOf('/') + 1);
+            secondSrc = secondSrc.substr(secondSrc.lastIndexOf('/') + 1);
+            if (ev.currentTarget.id > 0 && ev.currentTarget.id <= 6 && tgt.parentNode.id > 0 && tgt.parentNode.id <= 6) {
+                this.currArrState[Number(ev.currentTarget.id)] = firstSrc;
+                this.currArrState[Number(tgt.parentNode.id)] = secondSrc;
+            }
+
+        } else {
+            ev.currentTarget.appendChild(data);
+            let firstSrc = ev.currentTarget.firstElementChild.src;
+            firstSrc = firstSrc.substr(firstSrc.lastIndexOf('/') + 1);
+            if (ev.currentTarget.id > 0 && ev.currentTarget.id <= 6) {
+                this.currArrState[Number(ev.currentTarget.id)] = firstSrc;
+            }
+        }
+
+        let f = false;
+        let arrLength = 0;
+        for (let i = 0; i < this.solnArr.length; i++) {
+            if (this.currArrState[i + 1] !== undefined) {
+                arrLength++;
+            }
+            if (this.currArrState[i + 1] !== this.solnArr[i]) {
+                f = true;
```

Comment:
would there ever be a case such that this if statement is true when `i = 0`, so `f` now is `true`. then the loop continues, so `arrLength` eventually becomes 6.

ps: `f` is an awful variable name